### ISSUE TITLE
amplify auth update CloudFront still In Progress

### ISSUE THREAD

**Issue by michaelcuneo** [aws-amplify/amplify-cli#842]
Not sure if ... Bug, or something else.

**To Reproduce**
1. amplify hosting update
2. Added CloudFront configuration.

Started at 10:03 this morning, still says In Progress 23 minutes later.

**Comment by yuth** [comment]
@michaelcuneo Amplify CLI uses cloud formation to provision resources. And provisioning CloudFront takes quite a bit of time (usually around 30+ mins). This happens only once when you provision it for the first time for an  environment and subsequent pushes should be pretty quick.

Did you get any error while pushing? Or is it just slow?

**Comment by michaelcuneo** [comment]
@jordanranz @yuth It did end up completing, I was under the impression that initial CloudFront propagation took quite a while, but future updates were quite fast. For some reason this one took almost 45 minutes. It was already an active rolled out CloudFront, that only needed a minor update. So I didn't really expect it to take so long.

**Comment by michaelcuneo** [comment]
I did have some work to present the morning of this issue, but because of the massive CloudFront delay I had to postpone. All good, I'll make sure I have way longer before pushing updates in the future.
